core/arch/arm/plat-k3/drivers/sa2ul.c

```diff
@@ -33,8 +33,67 @@
 #define FW_SECURE_ONLY          GENMASK_32(7, 0)
 #define FW_NON_SECURE           GENMASK_32(15, 0)
 
+#define DDR16SS0_TI_SCI_FW_ID	1
+
 register_phys_mem_pgdir(MEM_AREA_IO_SEC, SA2UL_BASE, SA2UL_REG_SIZE);
 
+static __maybe_unused int protect_opee_memory(void)
+{
+	uint8_t owner_privid = 0;
+	uint8_t owner_index = OPTEE_HOST_ID;
+	uint16_t owner_permission_bits = 0;
+	uint16_t fwl_id = DDR16SS0_TI_SCI_FW_ID;
+	unsigned int region = 0;
+	uint32_t control;
+	uint32_t permissions[FWL_MAX_PRIVID_SLOTS] = { };
+	uint64_t start_address;
+	uint64_t end_address;
+	int ret = 0;
+
+	ret = ti_sci_change_fwl_owner(fwl_id, region, owner_index,
+				      &owner_privid, &owner_permission_bits);
+	if (ret) {
+		EMSG("Cannot set owner %d/%d\n", fwl_id, region);
+		return ret;
+	}
+
+	ret = ti_sci_get_fwl_region(fwl_id, region, 1,
+				    &control, permissions,
+				    &start_address, &end_address);
+	if (ret) {
+		EMSG("Could not get firewall region information");
+		return ret;
+	}
+
+	/* Modify firewall to allow all others access */
+	control = FW_ENABLE_REGION | FW_BACKGROUND_REGION;
+	start_address = DRAM0_BASE;
+	end_address = (uint64_t)DRAM0_BASE + DRAM0_SIZE - 1;
+	permissions[0] = (FW_WILDCARD_PRIVID << 16) | FW_NON_SECURE;
+	ret = ti_sci_set_fwl_region(fwl_id, region, 1,
+				    control, permissions,
+				    start_address, end_address);
+	if (ret) {
+		EMSG("Could not set firewall region information");
+		return ret;
+	}
+
+	/* Modify firewall to block all others access */
+	control = FW_ENABLE_REGION;
+	start_address = CFG_TZDRAM_START;
+	end_address = (uint64_t)CFG_TZDRAM_START + CFG_TZDRAM_SIZE - 1;
+	permissions[0] = (FW_WILDCARD_PRIVID << 16) | FW_SECURE_ONLY;
+	ret = ti_sci_set_fwl_region(fwl_id, 1, 1,
```

Comment:
Why not use SA2UL_TI_SCI_FW_RGN_ID here?

Reply:
Ok this is not the same firewall but explicit defines would help in this case (regions IDs)